src/js/api.js

```diff
@@ -91,9 +108,9 @@ class API {
   deleteArticle(params={}) {
     return this._wrap(this.client({
       method: "DELETE",
-      path: `/article/${params.id}`
+      path: `/articles/${params._id}`
```

Comment:
https://github.com/mozilla-services/readinglist/pull/91 is going to change that I think.